lib/normalize_title_filter.js

```diff
@@ -107,26 +90,31 @@ module.exports = function(hyper, req, next, options, specInfo) {
             }
         }
         if (normalizeResult.getNamespace().isFile()
-                && siteInfo.sharedRepoRootURI
                 // Temporarily limit file descripiton redirects to the app,
                 // until https://phabricator.wikimedia.org/T130757 (VE
                 // redirect support) is resolved.
                 // TODO: Remove.
                 && /^WikipediaApp\//.test(req.headers['user-agent'])) {
-            return next(hyper, req).catch({ status: 404 }, function() {
-                // It's a file page and it might be in the shared repo.
-                // Redirect.
-                var redirectPath = req.uri + '';
-                redirectPath = redirectPath.substr(redirectPath.indexOf('v1') + 2);
-                redirectPath = siteInfo.sharedRepoRootURI + '/api/rest_v1'
-                    + redirectPath + getQueryString(req);
-                return {
-                    status: 302,
-                    headers: {
-                        location: redirectPath,
-                        'cache-control': options.redirect_cache_control || 'no-cache'
+            return next(hyper, req).catch({ status: 404 }, function(e) {
+                return mwUtil.getSiteInfo(hyper, req)
+                .then(function(siteInfo) {
+                    if (siteInfo.sharedRepoRootURI) {
+                        // It's a file page and it might be in the shared repo.
+                        // Redirect.
+                        var redirectPath = req.uri + '';
+                        redirectPath = redirectPath.substr(redirectPath.indexOf('v1') + 2);
+                        redirectPath = siteInfo.sharedRepoRootURI + '/api/rest_v1'
+                        + redirectPath + getQueryString(req);
+                        return {
+                            status: 302,
+                            headers: {
+                                location: redirectPath,
+                                'cache-control': options.redirect_cache_control || 'no-cache'
+                            }
+                        };
                     }
-                };
+                    throw e;
```

Comment:
[Minor / matter of taste]: I personally prefer to use an else branch in this kind of situation to make the exclusivity clearer. I had to scan back up to figure out that the early return prevents this throw for commons redirects, while I wouldn't have done the same with an else.